src/sentry/spans/consumers/process/factory.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import uuid
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping
+
+import msgpack
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.produce import Produce
+from arroyo.types import Commit, Message, Partition, Topic
+from confluent_kafka import Producer
+
+from sentry import quotas
+from sentry.models import Project
+from sentry.spans.span import SPAN_SCHEMA_VERSION, Span
+from sentry.utils import json, kafka_config
+from sentry.utils.arroyo import RunTaskWithMultiprocessing
+
+
+def _build_snuba_payload(payload: Mapping[str, Any]) -> MutableMapping[str, Any]:
+    project = Project.objects.get_from_cache(id=payload["project_id"])
+    organization = project.organization
+
+    span_dict: MutableMapping[str, Any] = {}
+    span_dict["description"] = payload["description"]
+    span_dict["exclusive_time_ms"] = int(payload.get("exclusive_time", 0))
+    span_dict["is_segment"] = not payload.get("parent_span_id")
+    span_dict["organization_id"] = organization.id
+    span_dict["parent_span_id"] = payload.get("parent_span_id", "0")
+    span_dict["project_id"] = payload["project_id"]
+    span_dict["retention_days"] = quotas.get_event_retention(organization=organization) or 90
+    span_dict["span_id"] = payload.get("span_id")
+    span_dict["span_status"] = payload["status"]
+    span_dict["tags"] = payload.get("tags")
+    span_dict["trace_id"] = str(uuid.UUID(payload["trace_id"]))
```

Comment:
this should just be the 16 byte hex string.

Reply:
https://github.com/getsentry/sentry-kafka-schemas/blob/main/examples/snuba-spans/1/basic_span.json

Reply:
Are you sure you meant to target the `trace_id` field with your comment? I'm assuming you're talking about the `span_id` instead?

Reply:
I meant `trace_id` . str(uuid) produces a string formatted with dashes. I don't know if clickhouse uses the same format. It's easier to just pass the raw hex number around.

Reply:
Oh, I see what you're saying.